### PR TITLE
Chore/gulp build process

### PR DESCRIPTION
**Pull request by settinghead**
- Gulp sass
- Gulp JS compile & uglify
- ES6 Babel integration for Karma and gulp
- Karma coveralls report
- ESLint, with dev watch support
- Update README file
- Static dev server with live reload support

### CODE REVIEW

**Comment by settinghead** [comment]
@alex-deaconu the build process is mostly finished except for E2E sample test. Please take a look and let me know if I've missed anything. Thanks!

**Comment by settinghead** [comment]
@alex-deaconu as per our dicussion,

- updated the dev process to incorporate a separate dev server
- modified `Viewer.html` to execute the new portion of JavaScript (the one compiled by Node.js)
- Live reload for the new portion of code

File: src/main/webapp/Viewer.html [hunk 1]
```diff
@@ -83,22 +83,23 @@
...
     <!--                                           -->
     <script type="text/javascript" language="javascript" src="../viewer/viewer.nocache.js"></script>
 
-	<script src="../scripts/jquery-1.10.2.min.js" type="text/javascript"></script>	
+	<script src="../scripts/jquery-1.10.2.min.js" type="text/javascript"></script>
 	<script type="text/javascript" language="javascript" src="../scripts/viewerScripts.js"></script>
 	<script type="text/javascript" language="javascript" src="../scripts/ext-logger.js"></script>
-	
+  <script type="text/javascript" language="javascript" src="//127.0.0.1:11111/livereload.js?snipver=1"></script>
+	<script type="text/javascript" language="javascript" src="//127.0.0.1:9999/viewer.js"></script>
```
Comment (alex-deaconu) [inline src/main/webapp/Viewer.html]: @settinghead I'm not sure what your intention is here, but these scripts should not appear in the Deployed version. Also, where is the compiled viewer.js in the deployed HTML? (https://gulp-build-process-dot-rvaviewer-test.appspot.com/). Thanks.

Comment (settinghead) [inline src/main/webapp/Viewer.html]: @alex-deaconu you are right, this is a quick and hackish effort to see if a separate static server works in the dev environment. I haven't yet accounted for production & deployment. Was going to do it next.

Comment (alex-deaconu) [inline src/main/webapp/Viewer.html]: @settinghead sounds good. Thanks.

**Comment by settinghead** [comment]
@alex-deaconu updated:

- Gulp now spins up gwt:compile and war:exploded (necessary to update the `target` directory) automatically when files change
- build process deploys to staging correctly (verified at https://gulp-build-process-dot-rvaviewer-test.appspot.com/)
- I moved `src-js` back to root level, because it was interfering with the GWT compilation process, and because the uncompiled source files no longer needs to be in `webapp`

Please take a look and see if everything works correctly. Thanks!

File: .gitignore [hunk 2]
```diff
@@ -38,8 +38,13 @@ target/
 src/main/java/META-INF/persistence.xml
 src/main/webapp/WEB-INF/classes/
 src/main/webapp/viewer/
+src/main/webapp/dist/
 bin/
 
+# Node generated files
+/dist
```
Comment (alex-deaconu) [inline .gitignore]: @settinghead I believe this should be removed as we moved the `dist` folder. Thanks.

File: config/dev.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+const configDev = {
+
+};
+
+export default configDev;
```
Comment (alex-deaconu) [inline config/dev.js]: @settinghead curious is this the config for the build or for the app? I would rather not get the two mixed up. Also, if it's for the build, I'm wondering if you can make a config process for the app (to be compiled in with the sources). Thanks.

Comment (settinghead) [inline config/dev.js]: @alex-deaconu it's meant for build. OK I'll make another process for app.

**Comment by alex-deaconu** [comment]
@settinghead double checked and the code compiles and tests correctly, verified the same with my branch as well. Local env watches also work as expected. 

However there are some issues that we already discussed, such as minification not working and the lack of a environment based config for the code. Those can be addressed in future PRs.

Looks good 👍 . Thanks!

**Pull request merged**